package proto_ff;

import "yd_fieldoptions.proto";


message elementelementattributeDesc
{
	optional int32 type = 1[(yd_fieldoptions.field_cname) = ""];
}

message elementelementmaterialDesc
{
	optional int64 id = 1[(yd_fieldoptions.field_cname) = ""];
}

message elementelement
{
	optional int64 ID = 1[(yd_fieldoptions.field_cname) = "元素ID"];
	optional string resource = 2[(yd_fieldoptions.field_cname) = "资源形象", (yd_fieldoptions.field_bufsize) = 32];
	optional int32 type = 3[(yd_fieldoptions.field_cname) = "类型"];
	optional int32 level = 4[(yd_fieldoptions.field_cname) = "等级"];
	optional int32 exp = 5[(yd_fieldoptions.field_cname) = "元素升级经验"];
	optional int64 skillID = 6[(yd_fieldoptions.field_cname) = "解锁技能ID"];
	repeated elementelementattributeDesc attribute = 7[(yd_fieldoptions.field_cname) = "基础属性ID", (yd_fieldoptions.field_arysize) = 2];
	repeated elementelementmaterialDesc material = 8[(yd_fieldoptions.field_cname) = "进阶材料ID", (yd_fieldoptions.field_arysize) = 3];
	repeated int64 fragmentID = 9[(yd_fieldoptions.field_cname) = "关联碎片ID"];
}


message Sheet_elementelement
{
	repeated elementelement elementelement_List = 1[(yd_fieldoptions.field_arysize)=1708];
}

message elementfragmentattributeDesc
{
	optional int32 type = 1[(yd_fieldoptions.field_cname) = ""];
}

message elementfragment
{
	optional int64 ID = 1[(yd_fieldoptions.field_cname) = "碎片ID"];
	optional int64 itemID = 2[(yd_fieldoptions.field_cname) = "物品ID"];
	optional int32 itemNum = 3[(yd_fieldoptions.field_cname) = "碎片使用上限"];
	repeated elementfragmentattributeDesc attribute = 4[(yd_fieldoptions.field_cname) = "基础属性ID", (yd_fieldoptions.field_arysize) = 3];
}


message Sheet_elementfragment
{
	repeated elementfragment elementfragment_List = 1[(yd_fieldoptions.field_arysize)=116];
}

message elementelementUI
{
	optional int32 type = 1[(yd_fieldoptions.field_cname) = "类型"];
	optional string name = 2[(yd_fieldoptions.field_cname) = "元素名字", (yd_fieldoptions.field_bufsize) = 32];
}


message Sheet_elementelementUI
{
	repeated elementelementUI elementelementUI_List = 1[(yd_fieldoptions.field_arysize)=108];
}
